### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by lchastanet**
Add some features

### CODE REVIEW

File: src/Controller/UserController.php [hunk 1]
```diff
@@ -88,16 +120,22 @@ public function createUser(User $user, ConstraintViolationList $violations)
...
+        if (!in_array("ROLE_SUPER_ADMIN", $security->getUser()->getRoles())) {
+            $loggedUser = $userRepository->findOneBy(["userName" => $security->getUser()->getUsername()]);
+
+            if ($loggedUser->getClient()->getId() != $user->getClient()->getId()) {
+                throw new AccessDeniedException("You don't have the rights for modifying this user.");
             }
 
-            throw new ResourceValidationException($message);
+            $user->setRoles(["ROLE_USER"]);
+        } else {
```
Comment (friexo) [inline src/Controller/UserController.php:137]: Codacy found an issue: [The method updateUser uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/friexo/p7-daphp-oc/pullRequest?prid=6361015)

**Comment by friexo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- src/Service/Paginator.php  2
- src/DataFixtures/AppFixtures.php  1
         

Complexity decreasing per file
==============================
+ src/Controller/ProductController.php  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/friexo/p7-daphp-oc/pullRequest?prid=6361015&bid=20646163)

File: src/Service/Paginator.php [hunk 2]
```diff
@@ -18,46 +19,76 @@ class Paginator
...
             $this->offset = (int) ceil($page * $this->limit);
+
+            if ($page == 2) {
+                $this->offset = $this->limit;
+            }
+        }
+
+        if (empty($filter)) {
+            $dataSet = $this->getDataSet();
+        } else {
```
Comment (friexo) [inline src/Service/Paginator.php:83]: Codacy found an issue: [The method getPage uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/friexo/p7-daphp-oc/pullRequest?prid=6361015)

File: src/Service/Paginator.php [hunk 3]
```diff
@@ -18,46 +19,76 @@ class Paginator
...
+            ->getQuery()
+            ->getResult();
     }
 
-    private function getDataSet()
+    private function getFilteredDataSet($filter)
     {
-        return $this->entityRepository->createQueryBuilder('e')
+        $field = array_keys($filter)[0];
+        $id = $filter[$field];
```
Comment (friexo) [inline src/Service/Paginator.php:44]: Codacy found an issue: [Avoid variables with short names like $id. Configured minimum length is 3.](https://app.codacy.com/gh/friexo/p7-daphp-oc/pullRequest?prid=6361015)

**Comment by friexo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- src/DataFixtures/AppFixtures.php  1
- src/Service/Paginator.php  2
         

Complexity decreasing per file
==============================
+ src/Controller/ProductController.php  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/friexo/p7-daphp-oc/pullRequest?prid=6361015&bid=20646163)

File: src/Controller/UserController.php (hunk 1 above)
Comment (friexo) [inline src/Controller/UserController.php:137]: Codacy found an issue: [The method updateUser uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/friexo/p7-daphp-oc/pullRequest?prid=6361015)

File: src/Service/Paginator.php (hunk 2 above)
Comment (friexo) [inline src/Service/Paginator.php:83]: Codacy found an issue: [The method getPage uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/friexo/p7-daphp-oc/pullRequest?prid=6361015)

**Comment by lchastanet** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- src/DataFixtures/AppFixtures.php  1
- src/Service/Paginator.php  1
         

Complexity decreasing per file
==============================
+ src/Controller/ProductController.php  -2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/friexo/p7-daphp-oc/pullRequest?prid=6361015&bid=20646163)

File: src/Controller/UserController.php (hunk 1 above)
Comment (lchastanet) [inline src/Controller/UserController.php:137]: Codacy found an issue: [The method updateUser uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/friexo/p7-daphp-oc/pullRequest?prid=6361015)

**Pull request merged**